l5kit/l5kit/data/filter.py

```diff
@@ -132,3 +113,57 @@ def filter_tl_faces_by_status(tl_faces: np.ndarray, status: str) -> np.ndarray:
         np.ndarray: traffic light faces array with only faces with that status
     """
     return tl_faces[tl_faces["traffic_light_face_status"][:, TL_FACE_LABEL_TO_INDEX[status]] > 0]
+
+
+def get_frames_slice_from_scenes(scene_a: np.ndarray, scene_b: Optional[np.ndarray] = None) -> slice:
+    """
+    Get a slice for indexing frames giving a start and end scene
+
+    Args:
+        scene_a (np.ndarray): the starting scene
+        scene_b (Optional[np.ndarray]): the ending scene. If None, then scene_a end will be used
+
+    Returns:
+        slice: a slice object starting from the first frame in scene_a to the last one in scene_b
+    """
+    frame_index_start = scene_a["frame_index_interval"][0]
+    if scene_b is None:
+        scene_b = scene_a
+    frame_index_end = scene_b["frame_index_interval"][1]
+    return slice(frame_index_start, frame_index_end)
+
+
+def get_agents_slice_from_frames(frame_a: np.ndarray, frame_b: Optional[np.ndarray] = None) -> slice:
```

Comment:
One could think of having a function get_named_slice_from_frames(frame_a, frame_b, slice_name), where slice_name is a str param containing eg 'agent_index_interval' - saves a bit of code, but not sure if it's better.

Reply:
yeah I though about it, but this implementation is nice because it completely hides the name of the fields (which are again magic strings xD). In the case we add more intervals we should definitely apply your suggestion though